src/psyclone/psyir/backend/fortran.py

```diff
@@ -601,34 +601,54 @@ def gen_decls(self, symbol_table, args_allowed=True):
             which does not have a GlobalInterface or LocalInterface as this \
             is not supported by this backend.
         :raises VisitorError: if args_allowed is False and one or more \
-                              argument declarations exist in symbol_table.
-        :raises VisitorError: if any symbols representing variables (i.e. \
-            not kind parameters) without an explicit declaration or 'use' \
-            are encountered.
+            argument declarations exist in symbol_table.
+        :raises VisitorError: if any symbols representing variables without \
+            a (local) explicit declaration or 'use' are encountered and there \
+            are no wildcard imports.
 
         '''
         declarations = ""
-
-        if not all([isinstance(symbol.interface, (LocalInterface,
-                                                  GlobalInterface))
-                    for symbol in symbol_table.symbols
-                    if isinstance(symbol, RoutineSymbol)]):
-            raise VisitorError(
-                "Routine symbols without a global or local interface are not "
-                "supported by the Fortran back-end.")
+        # Keep a record of whether we've already checked for any wildcard
+        # imports to save doing so repeatedly
+        wildcard_imports_checked = False
+        first_wildcard_import = None
+
+        routine_symbols = [symbol for symbol in symbol_table.symbols
+                           if isinstance(symbol, RoutineSymbol)]
+        for sym in routine_symbols:
+            if (not isinstance(sym.interface, (LocalInterface,
+                                               GlobalInterface)) and
+                    sym.name.upper() not in FORTRAN_INTRINSICS):
+                if not wildcard_imports_checked:
+                    first_wildcard_import = \
+                        symbol_table.get_first_wildcard_import()
+                    wildcard_imports_checked = True
+                if not first_wildcard_import:
```

Comment:
Unless it returns a boolean as suggested it should be "if first_wilcard_import is not None"

Reply:
It's a boolean now.